src/utils/SSR.ts

```diff
@@ -0,0 +1,96 @@
+import { NormalizedCacheObject } from 'apollo-cache-inmemory';
+import { ApolloClient } from 'apollo-client';
+import { IncomingMessage } from 'http';
+import get from 'lodash.get';
+import NextCookies from 'next-cookies';
+import { LAYOUT_QUERY } from '../gql/common/layoutQuery';
+import { ApolloQueryOptions } from '../types/ApolloQueryOptions';
+import { Cookies } from '../types/Cookies';
+import { GetServerSidePropsContext } from '../types/GetServerSidePropsContext';
+import { PublicHeaders } from '../types/PublicHeaders';
+import { UniversalSSRPageProps } from '../types/UniversalSSRPageProps';
+import { UserSemiPersistentSession } from '../types/UserSemiPersistentSession';
+import { prepareGraphCMSLocaleHeader } from './graphcms';
+import { createApolloClient } from './graphql';
+import { DEFAULT_LOCALE, resolveFallbackLanguage } from './i18n';
+import { fetchTranslations, I18nextResources } from './i18nextLocize';
+import UniversalCookiesManager from './UniversalCookiesManager';
+
+/**
+ * getCommonServerSideProps returns only part of the props expected in UniversalSSRPageProps
+ * To avoid TS issue, we omit those that we don't return, and add those necessary to the getServerSideProps function
+ */
+export type getCommonServerSidePropsResults = Omit<UniversalSSRPageProps, 'apolloState' | 'customer'> & {
```

Comment:
A `getXX` for a type naming looks weird to me, why not just `CommonServerSidePropsResults` ?

Reply:
It looks weird at first, but that's what the Next.js core team does with the framework, so, for the core Next.js functions I've followed their way. See https://nextjs.org/docs/basic-features/typescript#static-generation-and-server-side-rendering

Reply:
Ok, make sense, maybe capitalize it would feel more a type, rather than a function.

Reply:
You're absolutely right! Just did.